typescript/packages/planning-server/src/index.ts

```diff
@@ -33,18 +33,37 @@ const handler = async (request: Request): Promise<Response> => {
   }
 };
 
+const cache: Record<string, any> = {};
+
 async function handleCreateThread(
   system: string,
   message: string,
-  activeTools: Anthropic.Messages.Tool[]
+  activeTools: Anthropic.Messages.Tool[],
 ): Promise<Response> {
+  const cacheKey = `${system}:${message}`;
+
+  if (cache[cacheKey]) {
+    console.log(
+      "Cache hit!",
+      (cacheKey.slice(0, 20) + "..." + cacheKey.slice(-20)).replaceAll(
+        "\n",
+        "",
+      ),
+    );
+    return new Response(JSON.stringify(cache[cacheKey]), {
+      headers: { "Content-Type": "application/json" },
+    });
+  }
```

Comment:
Simple memory cache to re-use responses for the same request. Useful to save on token spend in dev since we send the same requests every hot-reload.